wnbd-client/client.cpp

```diff
@@ -0,0 +1,353 @@
+{
+    if (vm.count("command-name")) {
+        string command_name = vm["command-name"].as<string>();
+        return print_command_help(command_name);
+    }
+    print_commands();
+    return 0;
+}
+
+void get_help_args(
+    po::positional_options_description &positonal_opts,
+    po::options_description &named_opts)
+{
+    positonal_opts.add("command-name", 1);
+    named_opts.add_options()
+        ("command-name", po::value<string>(), "Command name.");
+}
+
+DWORD execute_list(const po::variables_map &vm)
+{
+    return CmdList();
+}
+
+void get_show_args(
+    po::positional_options_description &positonal_opts,
+    po::options_description &named_opts)
+{
+    positonal_opts.add("instance-name", 1);
+    named_opts.add_options()
+        ("instance-name", po::value<string>()->required(), "Disk identifier.");
+}
+
+DWORD execute_show(const po::variables_map &vm)
+{
+    return CmdShow(safe_get_param<string>(vm, "instance-name").c_str());
+}
+
+void get_map_args(
+    po::positional_options_description &positonal_opts,
+    po::options_description &named_opts)
+{
+    positonal_opts.add("instance-name", 1)
+                  .add("hostname", 1);
+    named_opts.add_options()
+        ("instance-name", po::value<string>()->required(), "Disk identifier.")
+        ("hostname", po::value<string>()->required(), "NBD server hostname.")
+        ("port", po::value<DWORD>()->default_value(10809),
+            "NBD server port number. Default: 10809.")
+        ("export-name", po::value<string>(),
+            "NBD export name, defaults to <instance-name>.")
+        ("skip-handshake", po::bool_switch(),
+            "Skip NBD handshake and jump right into the transmission phase. "
+            "This is useful for minimal NBD servers. If set, the disk size "
+            "and block size must be provided.")
+        ("disk-size", po::value<UINT64>(),
+            "The disk size. Ignored when using NBD handshake.")
+        ("block-size", po::value<UINT32>(),
+            "The block size. Ignored when using NBD handshake.")
+        ("read-only", po::bool_switch(), "Enable disk read-only mode.");
+}
+
+DWORD execute_map(const po::variables_map& vm)
+{
+    return CmdMap(
+        safe_get_param<string>(vm, "instance-name").c_str(),
+        safe_get_param<string>(vm, "hostname").c_str(),
+        safe_get_param<DWORD>(vm, "port"),
+        safe_get_param<string>(vm, "export-name").c_str(),
+        safe_get_param<UINT64>(vm, "disk-size"),
+        safe_get_param<UINT32>(vm, "block-size"),
+        safe_get_param<bool>(vm, "skip-handshake"),
+        safe_get_param<bool>(vm, "read-only"));
+}
+
+void get_unmap_args(
+    po::positional_options_description &positonal_opts,
+    po::options_description &named_opts)
+{
+    positonal_opts.add("instance-name", 1);
+    named_opts.add_options()
+        ("instance-name", po::value<string>()->required(), "Disk identifier.")
+        ("hard-disconnect", po::bool_switch(), "Perform a hard disconnect.")
```

Comment:
Should we add more information that this is a hazardous operation: i.e. Perform a hard disconnect (any pending operation will be discarded). ?